### PR TITLE
Upgrade rimraf from v3 to v5 

### PR DESCRIPTION
**Pull request by JacobMGEvans**
The upgrade provides a few performance benefits and reliability in different OS systems. see: https://github.com/isaacs/rimraf\#major-changes

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/wranglerjs-compat-webpack-plugin/src/index.ts [hunk 1]
```diff
@@ -258,24 +259,12 @@ export class WranglerJsCompatWebpackPlugin {
...
-			}
-		};
-		options !== undefined
-			? rimraf(pathToRemove, options, callback)
-			: rimraf(pathToRemove, callback);
-	});
+function rm(pathToRemove: string, options?: RimrafSyncOptions): void {
+	options !== undefined
+		? rimrafSync(pathToRemove, options)
+		: rimrafSync(pathToRemove);
```
[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply packages/wranglerjs-compat-webpack-plugin/src/index.ts:267]: I wasn't trying to refactor the whole thing lol

[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply packages/wranglerjs-compat-webpack-plugin/src/index.ts:267]: I will see about improving it though

**Review by rozenmd — APPROVED** [review]
Looks fine to me

File: packages/wranglerjs-compat-webpack-plugin/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply packages/wranglerjs-compat-webpack-plugin/src/index.ts]: Removed the wrapper, it was only used in one place

**Comment by penalosa** [comment]
I'm a bit concerned about the change in `glob` behaviour:
> Globbing requires the --glob CLI option or glob option property to be set. (Removed in 4.0 and 4.1, opt-in support added in 4.2.)
Do we rely on that anywhere?

**Comment by JacobMGEvans** [comment]
> I'm a bit concerned about the change in `glob` behaviour:
> 
> > Globbing requires the --glob CLI option or glob option property to be set. (Removed in 4.0 and 4.1, opt-in support added in 4.2.)
> 
> Do we rely on that anywhere?

Not that I could find. There is only really 2 places we use rimraf one is the direct script usage in Wrangler package.json and the 2nd is the ONLY one in the webpack package. @penalosa

**Review by penalosa — APPROVED** [review]

**Pull request merged**